### PR TITLE
Make DialogFragment notification fullscreen

### PR DESCRIPTION
**Pull request by Ayyanchira**
Set Android's No Action Bar Style theme to notification.

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppFragmentHTMLNotification.java [hunk 1]
```diff
@@ -84,7 +89,23 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         if (args != null) {
             htmlString = args.getString("html", null);
         }
-        setStyle(DialogFragment.STYLE_NO_FRAME, 0);
+        viewModel = ViewModelProviders.of(this).get(InAppNotificationViewModel.class);
+        loadAndRetrieveFromViewModel();
+    }
+
+    private void loadAndRetrieveFromViewModel() {
+        //Checking transient variables which go null after configuration changes.
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppFragmentHTMLNotification.java]: Definitely not a right way to determining if configuration changes has occurred. I am looking for it.

File: iterableapi/src/main/java/com/iterable/iterableapi/model/InAppNotificationViewModel.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import android.arch.lifecycle.ViewModel;
+
+import com.iterable.iterableapi.IterableHelper;
+import com.iterable.iterableapi.IterableInAppLocation;
+
+public class InAppNotificationViewModel extends ViewModel {
+    public String htmlString;
+    public String messageID;
+    public IterableHelper.IterableUrlCallback clickCallback;
+    public IterableInAppLocation location;
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/model/InAppNotificationViewModel.java]: May be we can add backgroundAlpha and padding here too. Not sure as they, along with htmlString and messageID stay intact even without model class.
Either we can store all variables in model class making it uniform or just store the transient data which tend to get destroyed on configuration changes

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppFragmentHTMLNotification.java [hunk 3]
```diff
@@ -84,7 +89,23 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
...
+        if (clickCallback == null) {
+            htmlString = viewModel.htmlString;
+            messageId = viewModel.messageID;
+            clickCallback = viewModel.clickCallback;
+            location = viewModel.location;
+        } else {
+            viewModel.htmlString = htmlString;
+            viewModel.messageID = messageId;
+            viewModel.clickCallback = clickCallback;
+            viewModel.location = location;
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppFragmentHTMLNotification.java]: Definitely not a right way to determining if configuration changes has occurred. I am looking for it.

File: iterableapi/build.gradle [hunk 4]
```diff
@@ -56,6 +56,7 @@ dependencies {
     androidTestImplementation 'androidx.test.uiautomator:uiautomator:2.2.0'
     androidTestImplementation 'com.squareup.okhttp3:mockwebserver:4.2.2'
     androidTestImplementation 'org.mockito:mockito-android:3.0.0'
+    implementation 'android.arch.lifecycle:extensions:1.1.1'
```
Comment (vbabenkoru) [inline iterableapi/build.gradle]: This adds an extra dependency apps would have to pull when using our SDK.. As a library, we should limit our dependencies unless absolutely necessary.

**Comment by Ayyanchira** [comment]
Resolved by adding the change to Branch [MOB-961]

**Pull request closed without merging**